cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/DefaultStoreTest.java

```diff
@@ -663,4 +669,70 @@ public void testCheckDeletedProceduresAndWorkflow () throws Exception {
     Assert.assertEquals(0, specsToBeDeleted.size());
   }
 
+  @Test
+  public void testAdapterMDSOperations() throws Exception {
+    Id.Namespace namespaceId = new Id.Namespace("testAdapterMDS");
+
+    Map<String, String> properties = ImmutableMap.of("frequency", "10m");
+    List<Source> sources = Lists.newArrayList(new Source("eventStream", co.cask.cdap.adapter.DataType.STREAM,
+                                                         ImmutableMap.of("prop1", "val1")));
+
+    List<Sink> sinks = Lists.newArrayList(new Sink("myAvroFiles", DataType.DATASET,
+                                                   ImmutableMap.of("type", "co.cask.cdap.data.dataset.Fileset")));
+
+    AdapterSpecification specStreamToAvro = new AdapterSpecification("streamToAvro", AdapterType.BATCH,
+                                                                     properties, sources, sinks);
+
+    AdapterSpecification specStreamToParquet = new AdapterSpecification("streamToParquet", AdapterType.BATCH,
+                                                                     properties, sources, sinks);
+
+    store.addAdapter(namespaceId, specStreamToAvro);
+    store.addAdapter(namespaceId, specStreamToParquet);
+
+    // Get non existing spec
+    AdapterSpecification retrievedSpec = store.getAdapter(namespaceId, "nonExistingAdapter");
+    Assert.assertNull(retrievedSpec);
+
+    //Retrieve specs
+    retrievedSpec = store.getAdapter(namespaceId, "streamToAvro");
+    // Check Adapter properties
+    Assert.assertEquals("streamToAvro", retrievedSpec.getName());
```

Comment:
if we had hashCode and equals defined for the classes we could just compare directly, and it could be useful elsewhere too.